packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -231,6 +293,39 @@ export class CfnInclude extends core.CfnElement {
     this.conditions[conditionName] = cfnCondition;
   }
 
+  private createNestedStack(nestedStackId: string, props?: CfnIncludeProps): void {
+    if (nestedStackId in this.template.Resources) {
+      if (this.template.Resources[nestedStackId].Type === 'AWS::CloudFormation::Stack' && props?.nestedStacks) {
+        const nestedStackScope = new core.NestedStack(this, 'Nested' + nestedStackId, {
+          parameters: this.template.Resources[nestedStackId].Properties.Parameters,
```

Comment:
Whoops! You forgot to translate these into CDK values using `CfnParser`!

That means we're missing quite some coverage here. Here's the test that I want you to write here.

```ts
  test('correctly handles renaming of references across nested stacks', () => {
    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
      templateFile: testTemplateFilePath('parent-template.json'),
      nestedStacks: {
        'ChildStack': {
          templateFile: testTemplateFilePath('doesn-matter.json'),
        },
      },
    });
    const cfnBucket = parentTemplate.getResource('Bucket');
    cfnBucket.overrideLogicalId('DifferentBucket');
    const parameter = parentTemplate.getParameter('Param');
    parameter.overrideLogicalId('DifferentParameter');

    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
      "Properties": {
        "Parameters": {
          "Param1": {
            "Ref": "DifferentParameter",
          },
          "Param2": {
            "Fn::GetAtt": ["DifferentBucket", "Arn"],
          },
        },
      },
    });
  });
```

Where `parent-template.json` (change the name to something better BTW) is:

```json
{
  "Parameters": {
    "Param": {
      "Type": "String"
    }
  },
  "Resources": {
    "Bucket": {
      "Type": "AWS::S3::Bucket"
    },
    "ChildStack": {
      "Type": "AWS::CloudFormation::Stack",
      "Properties": {
        "TemplateURL": "https://cfn-templates-set.s3.amazonaws.com/child-import-stack.json",
        "Parameters": {
          "Param1": { "Ref":  "Param" },
          "Param2": { "Fn::GetAtt":  ["Bucket", "Arn"] }
        }
      }
    }
  }
}
```

I believe it will fail today.

(Also - you're repeating an expression again)